crates/store/re_chunk_store/src/gc.rs

```diff
@@ -583,6 +584,13 @@ impl ChunkStore {
                                 }
                             }
 
+                            // We must keep track of the min Row IDs associated with these chunks too,
+                            // so that we don't have to rely on an O(n**2) `retain` call later.
+                            row_ids_removed.extend(chunk_ids.iter().filter_map(|chunk_id| {
+                                self.chunks_per_chunk_id
+                                    .get(chunk_id)
+                                    .and_then(|chunk| chunk.row_id_range().map(|(min, _)| min))
+                            }));
```

Comment:
I assume we have good reason to not remove from `self.chunk_ids_per_min_row_id` inplace because we still need that for bookkeeping in this loop and I just don't see it

Reply:
:notsure:

Reply:
that was just silly, nice catch

Reply:
oh nice. way better :D